server/src/main/java/org/elasticsearch/index/engine/ElasticsearchConcurrentMergeScheduler.java

```diff
@@ -117,7 +120,7 @@ protected void doMerge(MergeSource mergeSource, MergePolicy.OneMerge merge) thro
             beforeMerge(onGoingMerge);
             super.doMerge(mergeSource, merge);
         } finally {
-            long tookMS = TimeValue.nsecToMSec(System.nanoTime() - timeNS);
+            long tookMS = TimeValue.nsecToMSec(relativeTimeInNanosSupplier.getAsLong() - timeNS);
```

Comment:
I think the relative time supplier we use is the cached one and thus this degrades the took time into 200ms granularity. I am not sure it is important but I think I would prefer to keep this using an outside supplied `System.nanoTIme` instead.